### PR TITLE
send_summary

### PR DESCRIPTION
**Pull request by pg20**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -55,4 +55,21 @@ def create_token
       end while self.class.exists?(registration_token: registration_token)
     end
   end
+
+  def self.newtask(project)
```
Comment (DMips) [inline app/models/user.rb]: @pg20 why are they scopes? we want to find tasks for specific user so normal method would be sufficient

File: app/models/user.rb [hunk 2]
```diff
@@ -55,4 +55,21 @@ def create_token
...
+
+  def confirm_completed
+    User.project.tasks.where("(status = 2 and (creator_id = user.id and creator_id != finished_by) and updated_at IN (?) )", (DateTime.now - 24.hours)..DateTime.now)
+  end
+
+  def confirm_delete
+     User.project.tasks.where("(status = 4 and (creator_id = user.id and creator_id != finished_by) and updated_at IN (?) )", (DateTime.now - 24.hours)..DateTime.now)
+  end
+
 end
```
Comment (pg20) [inline app/models/user.rb]: @DMips now check :)

File: app/models/user.rb [hunk 3]
```diff
@@ -55,4 +55,21 @@ def create_token
...
+  end
+
+  def confirm_completed
+    User.project.tasks.where("(status = 2  and creator_id != finished_by and updated_at BETWEEN ? AND ? )", (DateTime.now - 24.hours), DateTime.now)
+  end
+
+  def confirm_delete
+     User.project.tasks.where("(status = 4 and creator_id != finished_by and updated_at BETWEEN ? AND ? )", (DateTime.now - 24.hours), DateTime.now)
+  end
+
```
Comment (pg20) [inline app/models/user.rb]: I have checked the queries are working now. :)

File: app/models/email_template.rb [hunk 4]
```diff
@@ -42,6 +43,8 @@ def users
       User.mentee.where.not(registration_token: nil)
     when :unregistered_mentors
       User.mentor.where.not(registration_token: nil)
+    when :daily_summary
+      User
```
Comment (pg20) [inline app/models/email_template.rb]: Is this correct ? It will be send to all users. Where will i call the query functions which return tasks ?

Comment (DMips) [inline app/models/email_template.rb]: honestly i think we don't have to use email templates for that - will be much easier to use normal mailer.

File: app/models/user.rb (hunk 3 above)
Comment (DMips) [inline app/models/user.rb]: well... they do work but we can simplify them - I'll do that later today ;)

File: app/models/email_template.rb (hunk 4 above)
Comment (pg20) [inline app/models/email_template.rb]: Okk its great then. It is easier to implement for me using simple mailers. Ill complete using by creating mailers :)

File: app/models/user.rb (hunk 3 above)
Comment (pg20) [inline app/models/user.rb]: Hi @DMips I am running from console : User.find(4).newtask
Is says undefined project. Why is to so ?

File: app/mailers/project_summary_mailer.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+class ProjectSummaryMailer < ApplicationMailer
+
+  def project_summary(user)
+    @user = user
+    mail(to: user.email, subject: "[Learn IT, Girl] Daily summary of your board", sent: DateTime.now)
```
Comment (pg20) [inline app/mailers/project_summary_mailer.rb]: I wanted a subject like this : "Subject: [Learn IT, Girl] Daily summary of your board <dd/mm>", Which has a date in subject. The above implementation is fine or you want me to change ?

Comment (DMips) [inline app/mailers/project_summary_mailer.rb]: what's this sent for?

Comment (pg20) [inline app/mailers/project_summary_mailer.rb]: Umm I dont know how to add date in subject :P SO i used this. It display the date on which the mail is sent.

File: app/views/project_summary_mailer/project_summary.html.slim [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  - unless @user.unconfirmed_deleted_tasks.empty?
+    |PENDING COMPLETED TASKS CONFIRMATIONS
+    - @user.unconfirmed_deleted_tasks.each do |delete_confirm|
+        = delete_confirm.title
+
+p
+  |Best regards,
+  br
+  i
+    | The "Learn IT, Girl" team
```
Comment (pg20) [inline app/views/project_summary_mailer/project_summary.html.slim]: @DMips In the ouput of the email, The first task appeared in the same line. How to bring it on next line.

File: app/mailers/project_summary_mailer.rb (hunk 5 above)
Comment (DMips) [inline app/mailers/project_summary_mailer.rb]: sorry read you comment :P anyway if you want to put date in subject please do it like that "[Learn IT, Girl] Daily summary of your board - #{DateTime.now.strftime("%d %b %Y")}".

File: lib/tasks/project_summary.rake [hunk 7]
```diff
@@ -0,0 +1,11 @@
+namespace :project_summary do
+
+  desc "send daily summary of projects to registered users"
+  task daily_project_summary: :environment do
+       @users = User.all
```
Comment (pg20) [inline lib/tasks/project_summary.rake]: Here the user are all or DO u want me to filter users who havent registered or havent logged in once.

File: app/mailers/project_summary_mailer.rb (hunk 5 above)
Comment (pg20) [inline app/mailers/project_summary_mailer.rb]: Ok thanxs :)

File: app/views/project_summary_mailer/project_summary.html.slim [hunk 8]
```diff
@@ -0,0 +1,35 @@
+h3
+  = "Dear, #{@user.full_name}"
+p
+  | Here is a summary of the changed in your board
+p
+  - unless @user.new_tasks.empty?
+    |NEW TASKS 
+    - @user.new_tasks.each do |newtask|
+        = newtask.title
```
Comment (DMips) [inline app/views/project_summary_mailer/project_summary.html.slim]: not sure what would look better so you can either add br after line 9 (in loop) or

```ruby
 - unless @user.new_tasks.empty?
     p NEW TASKS 
     - @user.new_tasks.each do |newtask|
     p = newtask.title
```

File: lib/tasks/project_summary.rake (hunk 7 above)
Comment (pg20) [inline lib/tasks/project_summary.rake]: @DMips here.

File: app/models/user.rb [hunk 9]
```diff
@@ -55,4 +76,5 @@ def create_token
       end while self.class.exists?(registration_token: registration_token)
     end
   end
+
 end
```
Comment (pg20) [inline app/models/user.rb]: Change query to task within week

File: lib/tasks/project_summary.rake [hunk 10]
```diff
@@ -0,0 +1,11 @@
+namespace :project_summary do
+
+  desc "send daily summary of projects to registered users"
+  task daily_project_summary: :environment do
+       @users = User.where(role: [2,3])
```
Comment (pg20) [inline lib/tasks/project_summary.rake]: change users, and dont send empty mail check

**Pull request merged**